test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,152 @@
+        // person with no private information
+        assertViewSuccess(addressBook, listWithAll, 1);
+
+        // person with some private information
+        assertViewSuccess(addressBook, listWithAll, 2);
+
+        // addressbook has more people than displayList
+        assertViewSuccess(addressBook, listWithSome, 1);
+    }
+
+    /**
+     * Creates a new View command.
+     */
+    private Command generateViewCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        ViewCommand command = new ViewCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+    
+    /**
+     * Creates a new ViewAll command.
+     */
+    private Command generateViewAllCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        Command command = new ViewAllCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Asserts that the details of the person at specific index can be successfully retrieved
+     * and displayed.
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               list.get(index - 1).getAsTextHidePrivate());
+        assertViewCommandBehaviour(addressBook, list, expectedMessage, index);
+        
+        expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                        list.get(index - 1).getAsTextShowAll());
+        assertViewAllCommandBehaviour(addressBook, list, expectedMessage, index);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+
+        assertViewCommandBehaviour(addressBook, list, expectedMessage, index);
+        assertViewAllCommandBehaviour(addressBook, list, expectedMessage, index);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * person not existing in the addressbook.
+     */
+    private void assertViewErrorPersonNotInAddressBook(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK;
+
+        assertViewCommandBehaviour(addressBook, list, expectedMessage, index);
+        assertViewAllCommandBehaviour(addressBook, list, expectedMessage, index);
+    }
+    
+    /**
+     * Asserts that the View command gives correct feedback information
+     */
+    private void assertViewCommandBehaviour(AddressBook addressBook, List<ReadOnlyPerson> list, String message, int index) {
```

Comment:
Reorder parameters so that the three input values come first? 
`message` -> `expectedMessage`